pkg/reconciler/reconciler.go

```diff
@@ -359,6 +367,7 @@ func WithPostHook(h hook.PostHook) Option {
 	}
 }
 
+// todo: why it is not used/called in the main.go? What is the user case? Why it would be required?
```

Comment:
The default helm-operator uses the default value mapper, which is a no-op mapper that returns the input CR spec values directly. However users using this repo as a go library may want to have a custom mapping that maps the CR spec to a completely different set of values that their helm charts know about.

A common use case that people have asked for is to take one CR value (e.g. `dbUser`) and set multiple chart values from it (e.g. `mysql.dbUser` and `webserver.dbUser`). That way the CR API presented to the user it nice and tidy.